QuizApp/QuizApp/Main/AppRouter.swift

```diff
@@ -44,6 +44,11 @@ class AppRouter: AppRouterProtocol {
         navigationController.pushViewController(solvingQuizViewController, animated: true)
     }
 
+    func showResults(with text: String) {
+        let quizResultViewController = container.resolve(QuizResultViewController.self, args: text)
+        navigationController.setViewControllers([quizResultViewController], animated: true)
```

Comment:
Push this VC. Also, Home VC should never be destroyed. Add a check to showHome - if HomeVC exists, pop to it, else create new HomeVC and set it.